File: musicShopProject/ClientApp/src/domain/address/address.ts
export class Address {
    constructor(
        public id: string,
        public city: string,
        public street: string,
        public home: string,
        public apartment: string,
        
    ) { }
}

export function toAddress(data: any): Address {
    return new Address(data.id, data.city, data.street, data.home, data.apartment);
}

export function toAddresses(data: any[]):Address[] {
    return data.map(toAddress);
}

File: musicShopProject/Model/Addresses/Address.cs
namespace musicShopProject.Model.Addresses
{
    public class Address
    {
        public Guid Id { get; }
        public String City { get; }
        public String Street { get; }
        public String Home {  get; }
        public String? Apartment { get; }

        public Address(Guid id, String city, String street, String home, String? apartment)
        {
            Id = id;
            City = city;
            Street = street;
            Home = home;
            Apartment = apartment;
        }
    }
}
